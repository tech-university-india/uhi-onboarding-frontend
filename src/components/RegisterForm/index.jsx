import React from "react";
import TextBox from "../TextBox";

const RegisterForm = () => {

  const [healthID,setHealthID] = React.useState('');

  const handleChange = (e) => {
    setHealthID(e.target.value);
  };

  return (
    <>
      <div className="mt-10">
        <TextBox placeholder="First Name" disabled={true}/>
      </div>
      <div className="mt-3">
        <TextBox placeholder="Middle Name" disabled={true} />
      </div>
      <div className="mt-3">
        <TextBox placeholder="Last Name" disabled={true}/>
      </div>
      <div className="mt-10">
        <TextBox placeholder="Email ID" />
      </div>
      <div className="mt-3">
        <TextBox placeholder="Password" />
      </div>
      <div className="mt-3">
        <TextBox placeholder="Health ID" partialText=".ayush@sbx" onChange={handleChange} value={healthID} />
      </div>
    </>
  );
};

export default RegisterForm;
