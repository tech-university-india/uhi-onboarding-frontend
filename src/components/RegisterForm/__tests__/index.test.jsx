import RegisterForm from "..";
import { fireEvent, render, screen } from "@testing-library/react";

describe("RegisterForm", () => {
    it("should render correctly", () => {
        const { asFragment } = render(<RegisterForm />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render the correct text", () => {
        render(<RegisterForm />);
        expect(screen.getByPlaceholderText("First Name")).toBeInTheDocument();
        expect(screen.getByPlaceholderText("Middle Name")).toBeInTheDocument();
    });

    it("should handle onChange field when user types input", () => {
        render(<RegisterForm />);
        const input = screen.getByPlaceholderText("Health ID");
        expect(input.value).toBe("");
        fireEvent.change(input, { target: { value: "1234567890" } });
        expect(input.value).toBe("1234567890");
    });
    
})