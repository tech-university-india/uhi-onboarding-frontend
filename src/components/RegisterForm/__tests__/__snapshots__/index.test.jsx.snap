// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm should render correctly 1`] = `
<DocumentFragment>
  <div
    class="mt-10"
  >
    <input
      class="disabled:text-preFillText p-3 w-full rounded-lg shadow-textBoxInset bg-textBox"
      disabled=""
      placeholder="First Name"
      type="text"
      value=""
    />
  </div>
  <div
    class="mt-3"
  >
    <input
      class="disabled:text-preFillText p-3 w-full rounded-lg shadow-textBoxInset bg-textBox"
      disabled=""
      placeholder="Middle Name"
      type="text"
      value=""
    />
  </div>
  <div
    class="mt-3"
  >
    <input
      class="disabled:text-preFillText p-3 w-full rounded-lg shadow-textBoxInset bg-textBox"
      disabled=""
      placeholder="Last Name"
      type="text"
      value=""
    />
  </div>
  <div
    class="mt-10"
  >
    <input
      class="disabled:text-preFillText p-3 w-full rounded-lg shadow-textBox bg-textBox"
      placeholder="Email ID"
      type="text"
      value=""
    />
  </div>
  <div
    class="mt-3"
  >
    <input
      class="disabled:text-preFillText p-3 w-full rounded-lg shadow-textBox bg-textBox"
      placeholder="Password"
      type="text"
      value=""
    />
  </div>
  <div
    class="mt-3"
  >
    <div
      class="relative"
    >
      <input
        class="disabled:text-preFillText p-3 w-full rounded-lg shadow-textBox bg-textBox text-right"
        data-testid="partialText"
        placeholder="Health ID"
        style="padding-right: 6.25rem;"
        type="text"
        value=""
      />
      <span
        class="absolute block right-3 top-3 z-10 text-preFillText"
      >
        .ayush@sbx
      </span>
    </div>
  </div>
</DocumentFragment>
`;
